### PR TITLE
ci: add GPT based code review

### PR DESCRIPTION
**Pull request by dciborow**
## Description

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

File: .github/workflows/on-pull-request-target-review.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 2
+      - id: review
+        uses: microsoft/gpt-review@v0.8.0
+        with:
+          GIT_COMMIT_HASH: ${{ github.event.pull_request.head.sha }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          // OPENAI_ORG_KEY: ${{ secrets.OPENAI_ORG_KEY }}
```
Comment (dciborow) [inline .github/workflows/on-pull-request-target-review.yml:18]:
```suggestion
```

File: .github/workflows/on-pull-request-target-review.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+on: [pull_request_target]
```
Comment (dciborow) [inline .github/workflows/on-pull-request-target-review.yml:1]:
```suggestion
on: [pull_request]
```

**Review by skydiver1980 — APPROVED** [review]

**Comment by ChrisSidebotham** [comment]
Dear @dciborow , Azure Verified Modules ([AVM](https://aka.ms/AVM)) will become the single Microsoft standard for Bicep modules. Transitioning to AVM, requests for new features or bug fixes will need to be submitted against the appropriate AVM module via [AVM Module Issues](https://aka.ms/AVM/Bicep/ModuleIssue). 

As part of this transition, the new AVM CI environment is replacing the existing CI formerly leveraged within the Bicep Registry Modules (BRM) repository. As a result of this, issues and features related to the existing BRM CI will be closed. If you think your issue still needs addressing, please raise a new issue related to the AVM CI, [here](https://aka.ms/BRM/AVMQuestionFeedback).

See this [informational notice](https://github.com/Azure/bicep-registry-modules?tab=readme-ov-file#%EF%B8%8F-upcoming-changes-%EF%B8%8F) for more details on the upcoming changes.

Thank you for your understanding!

**Comment by ChrisSidebotham** [comment]
Closing as no follow up was provided

**Pull request closed without merging**